frontend/lib/identification/identification_qr_content_parser.dart

```diff
@@ -1,114 +1,88 @@
 import 'dart:convert';
-import 'dart:typed_data';
 
 import 'package:base32/base32.dart';
 
-import '../qr_code_scanner/qr_code_parser.dart';
+import '../qr_code_scanner/qr_code_processor.dart';
 import 'base_card_details.dart';
 import 'card_details.dart';
 import 'card_details_model.dart';
 import 'protobuf/card_activate_model.pb.dart';
 
-QRCodeParseResult createFieldMissingError(
-    String missingFieldName, String publicErrorCode) {
-  return QRCodeParseResult(
-      hasError: true,
-      userErrorMessage: "Beim Verarbeiten des eingescannten Codes ist ein"
-          "Fehler aufgetreten. Fehlercode: $publicErrorCode");
+class QRCodeInvalidTotpSecretException extends QrCodeParseException {
+  QRCodeInvalidTotpSecretException() : super("invalid totp secret");
+}
+
+class QRCodeInvalidExpiryException extends QrCodeParseException {
+  QRCodeInvalidExpiryException() : super("invalid expiry date");
+}
+
+class QRCodeMissingExpiryException extends QrCodeFieldMissingException {
+  QRCodeMissingExpiryException() : super("expirationDate");
+}
+
+class QRCodeInvalidFormatException extends QrCodeParseException {
+  QRCodeInvalidFormatException() : super("invalid format");
 }
 
 class IdentificationQrContentParser {
   final CardDetailsModel _cardDetailsModel;
 
   IdentificationQrContentParser(this._cardDetailsModel);
 
-  QRCodeParseResult processQrCodeContent(String rawBase64Content) {
+  void processQrCodeContent(String rawBase64Content) {
+    final base64Decoder = Base64Decoder();
+
+    CardActivateModel cardActivateModel;
     try {
-      final base64Decoder = Base64Decoder();
+      var rawProtobufData = base64Decoder.convert(rawBase64Content);
+      cardActivateModel = CardActivateModel.fromBuffer(rawProtobufData);
+    } on Exception catch (_) {
+      throw QRCodeInvalidFormatException();
+    }
```

Comment:
Just wondering, what kind of stuff could happen during protobuf parsing? Could it be worth it printing out an error message to the console?

Reply:
Good point, I will add that 👍🏽 
- [ ] log error

Reply:
Minor: Is it a best practice to include "inner exceptions" or is that bad style? :see_no_evil:

Reply:
In my opinion, it is good style to do so when the inner exception carries relevant information 👍🏽